python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -138,7 +138,7 @@ def llama_attention_forward_4_31(
         fsdp_flag = check_flash_attention_available(hidden_states)
     else:
         fsdp_flag = False
-    if fsdp_flag and q_len > 1:
+    if fsdp_flag and q_len > 1 and self.q_proj.qtype != ggml_tensor_qtype["bf16"]:
```

Comment:
we should have `BF16Linear` to deal with the input type instead

Reply:
Yes,  will update related logic in `BF16Linear`.
But at this code place, if we don't add some limitation, here `attention_dtype` will be set to fp16 for first token on pvc.
Even if we can cast fp16 into bf16 inside `BF16Linear`, I feel it's strange to cast q,k,v Tensor from bf16 to fp16 then use flash attention, and then cast fp16 to bf16 inside `o_proj`.
I perfer here just keep `attention_dtype` as bfloat16.
How about change
`if fsdp_flag and q_len > 1 and self.q_proj.qtype != ggml_tensor_qtype["bf16"]:`
to
`if fsdp_flag and q_len > 1 and original_dtype in [torch.float32, torch.float16]:`